src/python/sortingview/workspace_list/workspace_list.py

```diff
@@ -0,0 +1,31 @@
+from typing import Any
+import kachery_client as kc
+from ..workspace import Workspace
+
+def get_workspace_list(name: str='default'):
+    x = kc.get({'type': 'sortingview-workspace-list', 'name': name})
+    if x is None: x = []
+    return x
+
+def set_workspace_list(workspace_list, *, name: str='default'):
+    if not isinstance(workspace_list, list): raise Exception('Not a list')
+    for w in workspace_list:
+        if not isinstance(w, dict): raise Exception('Invalid workspace in list')
+        if not isinstance(w.get('workspaceUri', None), str): raise Exception('Missing or invalid field in workspace: workspaceUri')
+        if not isinstance(w.get('label', None), str): raise Exception('Missing or invalid field in workspace: label')
+        if not isinstance(w.get('metaData', {}), dict): raise Exception('Invalid field in workspace: metaData')
+        for k in w.keys():
+            if k not in ['workspaceUri', 'label', 'metaData']:
+                raise Exception(f'Invalid field in workspace: {k}')
```

Comment:
Ooh, basically a type guard for `Workspace`s! (or for `workspace_list` entries anyway...) These few tests are really short and clear, but I'd consider popping them out to their own function nonetheless, in case we want to reuse them & to make it obvious that they're basically defining a precise structure for the data.

It looks like Python does have native support for type guards (which I did not know until googling it just now)--it's [PEP647](https://www.python.org/dev/peps/pep-0647/). We could do something like this if we wanted to get really fancy, but I could also see an argument that this is a rabbit hole.

Reply:
UPDATE: Okay, I got too excited about the type guard thing; looking more closely I see that `Workspace` is a full-on object with actual logic, so we can't just cast a `dict` to it. I do still think this should be functionalized if possible, though.